shadows/framework/src/main/java/org/robolectric/shadows/ShadowParcelFileDescriptor.java

```diff
@@ -163,4 +223,80 @@ interface ParcelFileDescriptorReflector {
     @Direct
     void close();
   }
+
+  /**
+   * Tracks all the opened files opened by {@link ShadowParcelFileDescriptor#open(File, int)}.
+   *
+   * <p>Only for internal used by PFD.
+   *
+   * <p>The concept of "OpenedFileTable" is quite similar to *nix.
+   *
+   * <p>Note that is not supposed to be run on Windows, for the key `fd` is only available in *nix.
+   * In Windows the `handle` should be used instead of the `fd`.
+   */
+  private static class OpenedFileTable {
+    private final Map<FileDescriptor, OpenedFileItem> openedFileItems = new WeakHashMap<>();
+
+    void open(RandomAccessFile file, int mode, int fd) throws IOException {
+      synchronized (openedFileItems) {
+        openedFileItems.put(file.getFD(), new OpenedFileItem(file, mode, fd));
```

Comment:
Is it possible replace file item with the same fd? If possible, should we close old file with the same fd?